src/main/java/org/mule/extension/email/internal/sender/EmailBody.java

```diff
@@ -123,8 +123,13 @@ public String getOverrideEncoding() {
   /**
    * @return the body of the message content. The body aims to be text.
    */
-  public String getContentAsString(Charset charset) throws IOException {
-    return isNull(content) || isNull(content.getValue()) ? "" : IOUtils.toString(content.getValue(), charset);
+  public synchronized String getContentAsString(Charset charset) throws IOException {
```

Comment:
please add some unit tests that make sure this functionality works as expected

Reply:
Done